cli/src/main/java/denominator/cli/Denominator.java

```diff
@@ -309,6 +323,43 @@ public void run() {
       }
     }
 
+    private Function<Object, Object> attemptToTransformCredentials = new Function<Object, Object>() {
+      @Nullable
+      @Override
+      public Object apply(@Nullable Object input) {
+        input = attemptExtractX509Certificate(input);
+        return attemptExtractPrivateKey(input);
+      }
+    };
+
+    private Object attemptExtractX509Certificate(Object value) {
+      if (value != null && value instanceof String && value.toString().contains("BEGIN CERTIFICATE")) {
+        try {
+          X509CertParser x509CertParser = new X509CertParser();
+          x509CertParser.engineInit(new ByteArrayInputStream(value.toString().getBytes()));
+          return x509CertParser.engineRead();
+        } catch (Exception ex) {
+          return value;
+        }
+      }
+      return value;
+    }
+
+    private Object attemptExtractPrivateKey(Object value) {
```

Comment:
since you are using functions, you could make this and attemptExtractX509Certificate and attemptExtractPrivateKey functions. then make `decodeAnyPems() = Functions.compose(maybeDecodeX509Pem(), maybeDecodePrivateKeyPem());`

up to you as I don't feel strongly about this one.